0_RootFS/llvm_common.jl

```diff
@@ -1,3 +1,23 @@
+### Instructions for adding a new version
+#
+# * add the version/commit SHA to `llvm_tags`.  Commit SHAs can be found
+#   by viewing: https://github.com/llvm/llvm-project/releases
+# * create the directory `0_RootFS/LLVMBootstrap@X`.  You can copy the
+#   `build_tarballs.jl` file from `0_RootFS/LLVMBootstrap@X-1` and change the
+#   version to build.  Typically, to reduce patch duplication, we want to use
+#   symlinks if possible for patches.  Unfortunately, symlinks don't work
+#   with LLVMBootstrap so you'll need to copy all of the previous patches:
```

Comment:
Example of failure:
```
[09:19:12] Applying patch /workspace/srcdir/patches/0002-llvm8_libcxx_musl.patch
[09:19:12]  ---> atomic_patch -p1 ${f}
[09:19:12] Attempting to apply 0002-llvm8_libcxx_musl.patch...
[09:19:12] /usr/local/bin/atomic_patch: line 11: /workspace/srcdir/patches/0002-llvm8_libcxx_musl.patch: No such file or directory
[09:19:12] Patch 0002-llvm8_libcxx_musl.patch could not be applied! Reverting...
[09:19:12] /workspace/srcdir/patches/0002-llvm8_libcxx_musl.patch: No such file or directory
```

Reply:
We can get symlinks to the patches to work by upgrading `llvm_sources()`.

We haven't touched this file since we added the `DirectorySource`, `ArchiveSource`, etc... types.  The `DirectorySource` constructor has a kwarg `follow_symlinks` that you can set to `true` in order to get the proper content.